tools/language-server/astro.d.ts

```diff
@@ -0,0 +1,28 @@
+type AstroRenderedHTML = string;
+
+interface FetchContentResult {
+  astro: {
+    headers: string[];
+    source: string;
+    html: AstroRenderedHTML;
+  };
+  url: URL;
+  [key: string]: any;
+}
```

Comment:
It would be nice if `fetchContent` could be passed a generic to allow users to type their frontmatter (if they want to). This would fallback to `{ [key: string]: any }` if not provided. 

```suggestion
type FetchContentResult<ContentFrontmatter extends Record<string, any> = Record<string, any>> = {
  astro: {
    headers: string[];
    source: string;
    html: AstroRenderedHTML;
  };
  url: URL;
} & ContentFrontmatter;
```